src/isomorphic/ReactDebugTool.js

```diff
@@ -58,17 +63,40 @@ var ReactDebugTool = {
   onSetState() {
     emitEvent('onSetState');
   },
-  onMountRootComponent(internalInstance) {
-    emitEvent('onMountRootComponent', internalInstance);
+  onSetDisplayName(debugID, displayName) {
+    emitEvent('onSetDisplayName', debugID, displayName);
+  },
+  onSetIsEmpty(debugID, isEmpty) {
+    emitEvent('onSetIsEmpty', debugID, isEmpty);
```

Comment:
Same here. I'd prefer if we could get rid of exposing this limitation because my reconciler just won't output any nodes for something that is empty. So I won't have this concept nor this limitation.

So it is exposing an implementation detail.

> The tree devtool becomes unaware of parent-child association between composite and empty component so it can't clean it up on unmount easily

You can also just not expose it at all to the debug tool. That way it won't need to be cleaned up. Empty components doesn't have children or anything at all that is exposed.

So all you have to do is ignore any call to the debug tool that passes an instance of something "empty".